test/index.js

```diff
@@ -125,5 +125,20 @@ describe('KinesisStream', function() {
         done();
       });
     });
+
+    it('should putRecords without error when record contains circular references', function() {
+      const dataKeyFromSpy = (s) => { return s.getCall(0).args[0][0].Data; }
+      ks.putRecords = sinon.spy();
+
+      const message = {
+        hi: 'hello'
+      };
+      message.message = message;
+
+      ks.dispatch([message]);
+
+      expect(ks.putRecords.calledOnce).to.be.true;
+      expect('[Circular]').to.equal(dataKeyFromSpy(ks.putRecords));
```

Comment:
Shouldn't it be { hi: 'hello', message: '[Circular]' }?

Reply:
yes